### PR TITLE
HW-jv-fruit-shop-tests-22-Of-May

### PR DESCRIPTION
**Pull request by ViktoriiaSvidovska**

### CODE REVIEW

File: src/test/java/core/basesyntax/service/CsvFileReaderServiceImplTest.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    public void readDataFromFile_isOk() {
+        List<String> actual = csvFileReaderService
+                .readDataFromFile(configReader.getValueByKey("file.read.from"));
+        Assertions.assertEquals(EXPECTED_LIST, actual,
+                "The actual list doesn't match the expected list");
+    }
+
+    @Test
+    public void readDataFromFile_nullFilePath_isNotOk() {
+        Assertions.assertThrows(NullPointerException.class, () ->
```
Comment (aaananas) [inline src/test/java/core/basesyntax/service/CsvFileReaderServiceImplTest.java:59]: you should not expect NPE, remove this test

File: src/test/java/core/basesyntax/service/CsvFileReaderServiceImplTest.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+    @AfterEach
+    public void restoreFileCopy() throws IOException {
+        Path filePath = Path.of(configReader.getValueByKey("file.read.from"));
+        Files.copy(fileCopyPath, filePath, StandardCopyOption.REPLACE_EXISTING);
+        Files.delete(fileCopyPath);
+    }
+
+    @Test
+    public void readDataFromFile_isOk() {
```
Comment (aaananas) [inline src/test/java/core/basesyntax/service/CsvFileReaderServiceImplTest.java:50]: follow a pattern when naming your test methods
```<method name>_<short description>_<expected behavior>()```

File: src/test/java/core/basesyntax/service/CsvFileReportServiceImplTest.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+import java.util.Map;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class CsvFileReportServiceImplTest {
```
Comment (aaananas) [inline src/test/java/core/basesyntax/service/CsvFileReportServiceImplTest.java:19]: image

File: src/test/java/core/basesyntax/service/CsvFileReaderServiceImplTest.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        csvFileReaderService = new CsvFileReaderServiceImpl();
+        configReader = new ConfigReader();
+        randomDataGenerator = new RandomDataGenerator();
+    }
+
+    @BeforeEach
+    public void createFileCopy() throws IOException {
+        fileCopyPath = Path.of("src/test/resources", "file_copy.csv");
+        Files.copy(Path.of(configReader.getValueByKey("file.read.from")),
+                fileCopyPath, StandardCopyOption.REPLACE_EXISTING);
```
Comment (andriy-shulepa) [inline src/test/java/core/basesyntax/service/CsvFileReaderServiceImplTest.java:41]: you don't need to copy files. you created test file and you should use it. The problem is that you hardcoded path to your config inside ConfigReader. do refactoring so you will have separate configs for main code and for the test code

File: src/test/java/core/basesyntax/service/CsvFileWriterServiceImplTest.java [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+    public static void init() {
+        csvFileWriterService = new CsvFileWriterServiceImpl();
+        configReader = new ConfigReader();
+        randomDataGenerator = new RandomDataGenerator();
+    }
+
+    @BeforeEach
+    public void createFileCopy() throws IOException {
+        fileCopyPath = Path.of("src/test/resources", "file_copy.csv");
+        Files.copy(Path.of(configReader.getValueByKey("file.read.from")),
```
Comment (andriy-shulepa) [inline src/test/java/core/basesyntax/service/CsvFileWriterServiceImplTest.java:44]: same comment as for reader service

**Review by andriy-shulepa — CHANGES_REQUESTED** [review]

File: src/test/java/core/basesyntax/service/CsvFileWriterServiceImplTest.java [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+                        randomDataGenerator.generateRandomData()),
+                "NullPointerException expected");
+    }
+
+    public String createTestReport() {
+        CsvFileReportServiceImpl csvFileReportService = new CsvFileReportServiceImpl();
+        Map<String, Integer> allFruits = new HashMap<>();
+        allFruits.put("banana", 30);
+        allFruits.put("apple", 40);
+        return csvFileReportService.createReport(allFruits);
```
Comment (Rommelua) [inline src/test/java/core/basesyntax/service/CsvFileWriterServiceImplTest.java:119]: Why do you need this stuff here? Simple hardcode String with a few lines.

File: src/test/java/core/basesyntax/service/CsvFileWriterServiceImplTest.java [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+        } catch (IOException e) {
+            throw new RuntimeException("Can't write file:" + actualReportTest, e);
+        }
+    }
+
+    @Test
+    public void writeDataToFile_FileExist_Ok() {
+        Assertions.assertTrue(Files.exists(testReportPath.getParent()),
+                "The parent directory doesn't exist");
+        Assertions.assertTrue(Files.exists(testReportPath), "The file doesn't exist");
```
Comment (Rommelua) [inline src/test/java/core/basesyntax/service/CsvFileWriterServiceImplTest.java:63]: I do not see the call to `writeDataToFile()` in this test.

File: src/test/java/core/basesyntax/service/CsvFileWriterServiceImplTest.java [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+    }
+
+    @Test
+    public void writeDataToFile_InvalidFileName_NotOk() {
+        Assertions.assertThrows(RuntimeException.class,
+                () -> csvFileWriterService.writeDataToFile(INVALID_FILE_READ_FROM,
+                        randomDataGenerator.generateRandomData()),
```
Comment (Rommelua) [inline src/test/java/core/basesyntax/service/CsvFileWriterServiceImplTest.java:70]:
```suggestion
                       "test data"),
```
Remember KISS principle.

File: src/test/java/core/basesyntax/service/CsvFileWriterServiceImplTest.java [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+                "RuntimeException expected");
+    }
+
+    @Test
+    public void writeDataToFile_NullData_NotOk() {
+        Assertions.assertThrows(NullPointerException.class,
+                () -> csvFileWriterService
+                        .writeDataToFile(randomDataGenerator.generateRandomData(),
+                        null),
```
Comment (Rommelua) [inline src/test/java/core/basesyntax/service/CsvFileWriterServiceImplTest.java:102]: You should have a valid path when testing null data.

**Review by Rommelua — CHANGES_REQUESTED** [review]

**Review by olekskov — APPROVED** [review]
Good job

**Pull request closed without merging**